tindergames/src/test/java/org/teamcifo/logic/UserManagerTest.java

```diff
@@ -0,0 +1,164 @@
+        Faker faker = new Faker();
+        UserManager userManager = new UserManager();
+        User user = new User();
+        String fakeId = faker.number().digits(10);
+        user.setUserId(fakeId);
+        user.setName(faker.name().firstName());
+        user.setLastName(faker.name().lastName());
+        user.setEmail(faker.internet().emailAddress());
+        user.setPassword(faker.internet().password());
+
+        // Add the user to the user manager
+        userManager.addUser(user);
+
+        // Get the user from the user manager and check that it is the same as the user that was added
+        User retrievedUser = userManager.getUser(fakeId);
+        assertEquals(user, retrievedUser);
+    }
+
+    @Test
+    public void testAuthenticate() {
+        // Create an instance of the Faker library to generate fake data
+        Faker faker = new Faker();
+        UserManager userManager = new UserManager();
+        User user = new User();
+        String fakeId = faker.number().digits(10);
+        user.setUserId(fakeId);
+        user.setName(faker.name().firstName());
+        user.setLastName(faker.name().lastName());
+        user.setEmail(faker.internet().emailAddress());
+        String fakePassword = faker.internet().password();
+        user.setPassword(fakePassword);
+
+
+        // Add the user to the user manager
+        userManager.addUser(user);
+
+        // Authenticate the user and check that the result is true
+        boolean result = userManager.authenticate(fakeId, fakePassword);
+        assertTrue(result);
+
+        //
+        Assertions.assertTrue(userManager.authenticate(user.getUserId(), user.getPassword()));
+        Assertions.assertFalse(userManager.authenticate(user.getUserId(), "wrong password"));
+        Assertions.assertFalse(userManager.authenticate("non-existent user ID", "password"));
+    }
+
+    @Test
+    public void testGetAllUsers() {
+        // Create an instance of the Faker library to generate fake data
+        Faker faker = new Faker();
+        UserManager userManager = new UserManager();
+        Map<String,User> users = new HashMap<>();
+        for (int i = 0; i < 10; i++) {
+            User user = new User();
+            user.setUserId(faker.number().digits(10));
+            user.setName(faker.name().firstName());
+            user.setLastName(faker.name().lastName());
+            user.setEmail(faker.internet().emailAddress());
+            user.setPassword(faker.internet().password());
+            users.put(user.getUserId(),user);
+            userManager.addUser(user);
+        }
+
+        // Get the list of all users from the user manager and check that it is the same as the list of users that was created
+        Map<String,User> retrievedUsers = userManager.getAllUsers();
+        assertEquals(users,retrievedUsers);
+
+    }
+
+    //More tests
+    @BeforeEach
+    void setUp() {
+        userManager = new UserManager();
+        faker = new Faker();
+    }
```

Comment:
In java, tests aren't executed in order. If you define a @BeforeEach method, it will execute prior to any other test, so avoid repeating lines like `faker = new Faker();` inside the rest of the tests

Reply:
Code refactored, ignore previous comment. UserManager class is now static